msalautomationapp/src/androidTest/java/com/microsoft/identity/client/msal/automationapp/testpass/broker/TestCase833526Clone.java

```diff
@@ -0,0 +1,164 @@
+import com.microsoft.identity.client.ui.automation.interaction.PromptParameter;
+import com.microsoft.identity.client.ui.automation.interaction.UiResponse;
+import com.microsoft.identity.client.ui.automation.interaction.microsoftsts.AadPromptHandler;
+import com.microsoft.identity.internal.testutils.labutils.LabConfig;
+import com.microsoft.identity.internal.testutils.labutils.LabConstants;
+import com.microsoft.identity.internal.testutils.labutils.LabUserQuery;
+
+import org.junit.Test;
+
+import java.util.Arrays;
+
+// Broker Auth for MDM account
+// https://identitydivision.visualstudio.com/DevEx/_workitems/edit/833526
+@RetryOnFailure(retryCount = 2)
+@SupportedBrokers(brokers = BrokerCompanyPortal.class)
+public class TestCase833526Clone extends AbstractMsalBrokerTest {
+
+    @Test
+    public void test_833526() throws Throwable {
+        final String username = mLoginHint;
+        final String password = LabConfig.getCurrentLabConfig().getLabUserPassword();
+
+        final MsalSdk msalSdk = new MsalSdk();
+
+        final MsalAuthTestParams authTestParams = MsalAuthTestParams.builder()
+                .activity(mActivity)
+                .loginHint(username)
+                .scopes(Arrays.asList(mScopes))
+                .promptParameter(Prompt.SELECT_ACCOUNT)
+                .msalConfigResourceId(getConfigFileResourceId())
+                .build();
+
+        // start interactive token request in MSAL
+        // we expect to see enroll page as device is not enrolled
+        final MsalAuthResult authResult = msalSdk.acquireTokenInteractive(authTestParams, new OnInteractionRequired() {
+            @Override
+            public void handleUserInteraction() {
+                final PromptHandlerParameters promptHandlerParameters = PromptHandlerParameters.builder()
+                        .prompt(PromptParameter.SELECT_ACCOUNT)
+                        .loginHint(username)
+                        .sessionExpected(false)
+                        .consentPageExpected(false)
+                        .speedBumpExpected(false)
+                        .broker(mBroker)
+                        .expectingBrokerAccountChooserActivity(false)
+                        .enrollPageExpected(true)
+                        // cancel enroll here to short circuit as enroll will be started manually from CP anyway
+                        .enrollPageResponse(UiResponse.DECLINE)
+                        .build();
+
+                new AadPromptHandler(promptHandlerParameters)
+                        .handlePrompt(username, password);
+            }
+        }, TokenRequestTimeout.MEDIUM);
+
+        authResult.assertSuccess();
+
+        // enroll device with CP
+        final IMdmAgent mdmAgent = (IMdmAgent) mBroker;
+        mdmAgent.enrollDevice(username, password);
+
+        // SECOND REQUEST WITH LOGIN HINT
+        final TokenRequestLatch latchTryAcquireAgain = new TokenRequestLatch(1);
```

Comment:
I don't think the latch is needed anymore

Reply:
removed.